### ISSUE TITLE
Add function to retrieve Settings from Block

### ISSUE THREAD

**Issue by enzoc4** [input-output-hk/chain-wallet-libs#78]
Currently the only way to get a Settings object is with the `retrieve_funds` method, but the settings are needed for things not part of the registration process, like voting.

### Idea:
add a `Settings::from_block` constructor. And keep `retrieve_funds` as it is, so the new method could be used when it is known that the registration is already done.

**Comment by NicolasDP** [comment]
so `settings` can only be retrieved from the block0 and then it may be updated with a block.

The idea to have it done at the same time as the retrieve funds is that the process of going over the block0 to retrieve the funds require going through the whole block so if we can optimise it to apply the fragment one by one on both the settings and the wallet retrieval all the better.

**Comment by enzoc4** [comment]
That's not how the retrieve funds works at the moment, although we can make that optimization too. But isn't it a bit wasteful to do the whole address derivation-check every time for something that you do only once?

Something else we could do would be to provide a serialization method for the `Settings`, so they can be stored the first time and then just retrieved and used directly.
